### PR TITLE
docs(doc): Understanding filters

### PR DESCRIPTION
**Pull request by deven98**

### CODE REVIEW

File: docusaurus/docs/Flutter/guides/understanding_filters.mdx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+     Filter.in_('members', [user.id])
+])
+```
+
+#### FilterOperator.or
+
+Combines the provided filters and matches the values matched by at least one of the filters.
+
+```dart
+final filter = Filter.or([
```
Comment (funwithflutter) [inline docusaurus/docs/Flutter/guides/understanding_filters.mdx:128]: Should this be `Filter` or `FilterOperator`. The title says FilterOperator.or

File: docusaurus/docs/Flutter/guides/understanding_filters.mdx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+```
+
+### Group Queries
+
+#### FilterOperator.and
+
+The 'and' operator combines multiple queries.
+
+```dart
+final filter = Filter.and([
```
Comment (funwithflutter) [inline docusaurus/docs/Flutter/guides/understanding_filters.mdx:117]: Should this be `Filter` or `FilterOperator`. The title says FilterOperator.and

File: docusaurus/docs/Flutter/guides/understanding_filters.mdx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+     Filter.in_('shadowBannedUsers', [user.id])
+])
+```
+
+#### FilterOperator.nor
+
+Combines the provided filters and matches the values not matched by all the filters.
+
+```dart
+final filter = Filter.nor([
```
Comment (funwithflutter) [inline docusaurus/docs/Flutter/guides/understanding_filters.mdx:139]: Should this be `Filter` or `FilterOperator`. The title says FilterOperator.nor

File: docusaurus/docs/Flutter/guides/understanding_filters.mdx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+The 'notIn' filter allows getting objects where the key matches none in a specified array.
+
+```dart
+Filter.notIn('members', [user.id])
+```
+
+#### Filter.query
+
+Matches values by performing text search with the specified value.
```
Comment (funwithflutter) [inline docusaurus/docs/Flutter/guides/understanding_filters.mdx:88]:
```suggestion
The 'query' filter matches values by performing text search with the specified value.
```

File: docusaurus/docs/Flutter/guides/understanding_filters.mdx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+```dart
+Filter.query('name', 'demo')
+```
+
+#### Filter.autoComplete
+
+Matches values with the specified prefix.
```
Comment (funwithflutter) [inline docusaurus/docs/Flutter/guides/understanding_filters.mdx:96]:
```suggestion
The 'autoComplete' filter matches values with the specified prefix.
```

File: docusaurus/docs/Flutter/guides/understanding_filters.mdx [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+```dart
+Filter.autoComplete('name', 'demo')
+```
+
+#### Filter.exists
+
+Matches values that exist/don't exist based on the specified boolean value.
```
Comment (funwithflutter) [inline docusaurus/docs/Flutter/guides/understanding_filters.mdx:104]:
```suggestion
The 'exists' filter matches values that exist, or don't exist, based on the specified boolean value.
```

**Review by funwithflutter — CHANGES_REQUESTED** [review]

**Review by xsahil03x — APPROVED** [review]
LGTM

**Pull request merged**